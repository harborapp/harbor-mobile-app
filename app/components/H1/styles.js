import { StyleSheet } from 'react-native'

import * as global from '../../global-styles'

export default StyleSheet.create({
  header: {
    ...global.header,
    fontSize: 64
  }
})
