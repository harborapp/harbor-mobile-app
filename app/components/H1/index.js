import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'

import styles from './styles'

/**
 * H1 is a wrapper around the Text component to
 * style it in a way that it matches
 * an <h1> header.
 *
 * @returns {jsx}
 */
const H1 = ({ children, style, ...props }) => (
  <Text style={[styles.header, style]} {...props}>{children}</Text>
)

H1.propTypes = {
  children: PropTypes.element.isRequired,
  style: Text.propTypes.style
}

export default H1
