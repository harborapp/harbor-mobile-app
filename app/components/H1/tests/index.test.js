/* eslint-env jest */
import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import H1 from '../'

describe('H1', () => {
  it('should render', () => {
    const tree = renderer.create(
      <H1 id='hello'>
        <Text> Hello world! </Text>
      </H1>
    )

    expect(tree).toMatchSnapshot()
  })
})
