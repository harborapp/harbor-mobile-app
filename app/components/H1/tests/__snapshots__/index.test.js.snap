// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 should render 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  id="hello"
  style={
    Array [
      Object {
        "fontSize": 64,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Hello world! 
  </Text>
</Text>
`;
